plugin/src/main/java/io/deephaven/plugin/js/JsPlugin.java

```diff
@@ -3,16 +3,145 @@
  */
 package io.deephaven.plugin.js;
 
+import io.deephaven.annotations.BuildableStyle;
 import io.deephaven.plugin.Plugin;
+import io.deephaven.plugin.PluginBase;
+import org.immutables.value.Value.Check;
+import org.immutables.value.Value.Default;
+import org.immutables.value.Value.Immutable;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
 
 /**
- * A js plugin is a {@link Plugin} that allows adding javascript code under the server's URL path "js-plugins/". See
- * <a href="https://github.com/deephaven/deephaven-plugins#js-plugins">deephaven-plugins#js-plugins</a> for more details
- * about the underlying construction for js plugins.
+ * A JS plugin is a {@link Plugin} that allows for custom javascript and related content to be served, see
+ * {@link io.deephaven.plugin.js}.
+ *
+ * <p>
+ * For example, if the following JS plugin was the only JS plugin installed
+ *
+ * <pre>
+ * JsPlugin.builder()
+ *         .name("foo")
+ *         .version("1.0.0")
+ *         .main(Path.of("dist/index.js"))
+ *         .path(Path.of("/path-to/my-plugin"))
+ *         .build()
+ * </pre>
+ *
+ * the manifest served at "js-plugins/manifest.json" would be equivalent to
+ *
+ * <pre>
+ * {
+ *   "plugins": [
+ *     {
+ *       "name": "foo",
+ *       "version": "1.0.0",
+ *       "main": "dist/index.js"
+ *     }
+ *   ]
+ * }
+ * </pre>
  *
- * @see JsPluginPackagePath
- * @see JsPluginManifestPath
+ * and the file "/path-to/my-plugin/dist/index.js" would be served at "js-plugins/foo/dist/index.js". All other files of
+ * the form "/path-to/my-plugin/{somePath}" will be served at "js-plugins/foo/{somePath}".
```

Comment:
Seems kind of redundant - why not just say "All files of from "/path-to/my-plugin/{somePath}" will be served at "js-plugins/foo/{somePath}"? The `main` by definition is going to be relative to that path, ergo must always be served.